### PR TITLE
Disables Deprecation and Vulnerability info on Transitive packages

### PR DESCRIPTION
**Pull request by dominoFire**
## Bug

Fixes: https://github.com/NuGet/Client.Engineering/issues/1496

Regression? Last working version:

## Description

- Disables Deprecation and Vulnerability info on Transitive packages in PM UI Packages list when a package item view model tries to update package status.
- When Details pane is shown, a call to get package metadata is made to get all package details. This call also fetches Deprecation and Vulnerability metadata. Then, we just hide Deprecation and Vulnerability controls in Details Pane.
- Added two new properties to `PackageItemViewModel` to reflect when Deprecation and Vulnerability controls are visible.
- Added 2 new unit tests.

## PR Checklist

- [x] PR has a meaningful title
- [x] PR has a linked issue.
- [x] Described changes

- **Tests**
  - [x] Automated tests added: 2 new unit tests
  - ~[ ] Test exception~
  - ~[ ] N/A~

- **Documentation**
  - ~[ ] Documentation PR or issue filled~
  - [x] N/A: Vulnerability and deprecation is disable for first iteration.

## Validation

Visual Inspection

Before | After
----|----
PM With Deprecation | PM UI No Deprecation on Transitive Package
PM UI with Vulnerability | PM UI No Vulnerability on Transitive Package

Accessibility Insights FastPass:

Two a11y issues are shown. Those are not related with changes introduced in this PR.

FastPass results

FastPass issues

### CODE REVIEW

File: src/NuGet.Clients/NuGet.PackageManagement.UI/Models/DetailControlModel.cs [hunk 1]
```diff
@@ -385,6 +385,8 @@ protected void OnPropertyChanged(string propertyName)
 
         public bool IsPackageDeprecated => _packageMetadata?.DeprecationMetadata != null;
 
+        public bool IsDeprecationControlVisible => IsPackageDeprecated && _searchResultPackage.PackageLevel == PackageLevel.TopLevel;
```
Comment (jebriede) [inline src/NuGet.Clients/NuGet.PackageManagement.UI/Models/DetailControlModel.cs]: If we are now only loading metadata for top-level packages, then the vulnerability and deprecation data should not be there for this PackageItemViewModel and so IsPackageDeprecated and IsPackageVulnerable should be false, right? I'm not seeing why we'd need another boolean that checks again that the packages are top-level but I am probably missing something. Can you help me understand why we need the extra boolean for visibility?

**Comment by jebriede** [comment]
> When Details pane is shown, a call to get package metadata is made to get all package details. This call also fetches Deprecation and Vulnerability metadata. Then, we just hide Deprecation and Vulnerability controls in Details Pane.

@dominoFire Is this part of the PR description saying that the Details model still fetches the metadata but we just hide it for the details pane? If so, what we actually want to have happen is not to query the metadata at all to save on the network calls. Thanks in advance for clarifying!

**Comment by dominoFire** [comment]
> > When Details pane is shown, a call to get package metadata is made to get all package details. This call also fetches Deprecation and Vulnerability metadata. Then, we just hide Deprecation and Vulnerability controls in Details Pane.
> 
> @dominoFire Is this part of the PR description saying that the Details model still fetches the metadata but we just hide it for the details pane? If so, what we actually want to have happen is not to query the metadata at all to save on the network calls. Thanks in advance for clarifying!

I cannot totally disable fetching Metadata because the same network call is used for fetching package details

File: src/NuGet.Clients/NuGet.PackageManagement.UI/Models/DetailControlModel.cs (hunk 1 above)
Comment (dominoFire) [reply src/NuGet.Clients/NuGet.PackageManagement.UI/Models/DetailControlModel.cs]: It's just an easy fix to hide deprecation an vulnerability metadata. I now see you point on disabling metadata in details pane: I can not remove the network metadata call,  but I can stop 'calculating' vulnerability results in code.

Comment (jebriede) [reply src/NuGet.Clients/NuGet.PackageManagement.UI/Models/DetailControlModel.cs]: Locally installed packages are supposed to get their metadata from the installed package, right? Is that not happening for installed packages? Does it call to the registration API to get metadata?

Can we change how transitive dependencies are getting their metadata and not make network calls?

File: src/NuGet.Clients/NuGet.PackageManagement.UI/PackageItemLoader.cs [hunk 2]
```diff
@@ -312,6 +303,7 @@ public IEnumerable<PackageItemViewModel> GetCurrent()
                 }
                 else
                 {
+                    listItem.Project = _context.Projects.First();
```
Comment (dominoFire) [inline src/NuGet.Clients/NuGet.PackageManagement.UI/PackageItemLoader.cs:302]: Review: Try to remove Project if possible.

**Comment by dominoFire** [comment]
Closing since task is now closed.

**Pull request closed without merging**